components/molecule/select/src/index.js

```diff
@@ -71,18 +71,40 @@ const MoleculeSelect = props => {
     }
   )
 
-  useEffect(() => {
-    setOptionsData(getOptionData(children))
-  }, [children])
-
   const closeList = ev => {
     onToggle(ev, {isOpen: false})
     ev.preventDefault()
     ev.stopPropagation()
   }
 
+  const handleClick = ev => {
+    if (disabled) return
+    if (!refMoleculeSelect.current.contains(ev.target)) {
+      // outside click
+      closeList(ev)
+      setFocus(false)
+    } else {
+      refMoleculeSelect.current.focus()
+      setFocus(true)
```

Comment:
This `handleClick` is a callback for an event listener in document. So if MoleculeSelect does not contains the target you always focus MoleculeSelect. IMHO this is a bad practice because user possibly does not want to interact with this component always that him/her makes a mousedown or touchend.

Reply:
If `MoleculeSelect` or its children don't contain the event target it loses the focus and closes option list. If `MoleculeSelect` contains the target it gets focused. This is a common way to check if you're clicking outside the element.